src/components/search/SearchService.js

```diff
@@ -100,31 +110,37 @@ goog.provide('ga_search_service');
           position =
             [left > right ? left : right,
               right < left ? right : left];
-          if (ol.extent.containsCoordinate(
-              extent, position)) {
-            valid = true;
+          if (ol.extent.containsCoordinate(extent, position)) {
+            defer.resolve(roundCoordinates(position));
           } else {
-            position = ol.proj.transform(position,
-              'EPSG:2056', 'EPSG:21781');
-            if (ol.extent.containsCoordinate(
-                extent, position)) {
-              valid = true;
-            } else {
-              position =
-                [left < right ? left : right,
-                  right > left ? right : left];
-              position = ol.proj.transform(position,
-                'EPSG:4326', 'EPSG:21781');
-              if (ol.extent.containsCoordinate(
-                extent, position)) {
-                valid = true;
+            gaReframe.get95To03(position).then(function(response) {
```

Comment:
But then it is not self contained anymore? I need to put the code here-under in searchdirective. not sure I understand. the idea is to always return a promise with the final coordinates. so that you simply need to do
` gaSearchGetCoordinate(extent, query).then(function(coords) {});` and you're ensured you can use the `coords directly`